### PR TITLE
Added a new use case for transport.port

### PR DESCRIPTION
**Pull request by caminsha**
I've had the problem that I was allowing only the http.port setting and I couldn't get a connection..

After I tried using the transport port I had a successful connection and the sync work perfectly. 

On Wednesday I can show the issue that I tried it on the other port..

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

### CODE REVIEW

**Review by ywelsch — CHANGES_REQUESTED** [review]
Thank you for your contribution. I think it makes sense to name our Java transport client explicitly here. @jrodewig can you make a suggestion?

We can't account for other clients that wrap our Java transport client though, so I think it might be more appropriate if the docs of those clients (e.g. Cloudian Sync for Object Storage Integration) would contain that info.

Finally, as the transport client is going away in 8.0, this PR should be opened against the 7.x branch.

File: docs/reference/modules/transport.asciidoc
Comment (jrodewig) [inline docs/reference/modules/transport.asciidoc]:
```suggestion
the data). The transport module is also used for the `TransportClient` in the
{es} Java API.
```

**Review by jrodewig — CHANGES_REQUESTED** [review]
Thank you for your contribution @caminsha. 

I made some suggestions in line with @ywelsch's review.

Please note that you'll need to sign the Elastic CLA before we can merge this change.

Please let me know if you have any questions or concerns.

**Comment by caminsha** [comment]
@jrodewig @ywelsch  Thanks for your tips!

I've signed the CLA. I hope it is okay with the CLA otherwise contact me, thanks!

File: docs/reference/modules/transport.asciidoc
Comment (jrodewig) [inline docs/reference/modules/transport.asciidoc]:
```suggestion

```

Comment (jrodewig) [reply docs/reference/modules/transport.asciidoc]: Sorry. We'll need to remove this as well.

Otherwise, this looks good to me.

**Review by ywelsch — APPROVED** [review]
LGTM. The transport client is gone in 8.0, so this just needs to be merged in to 7.x and below. @jrodewig can you do this? Thank you

**Comment by ywelsch** [comment]
@elasticmachine test this please

**Review by jrodewig — APPROVED** [review]

**Comment by jrodewig** [comment]
@ywelsch  Will do.

Thanks again for your contribution @caminsha. I'll merge this after the CI tests pass.

**Pull request merged**
